test/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	appsv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func main() {
+	kubeconfig := filepath.Join(
+		os.Getenv("HOME"), ".kube", "config",
+	)
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		log.Fatal(err)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// NOTE: Certain features are in different clientsets. For example,
+	// Pods would be in CoreV1, not AppsV1
+	api := clientset.AppsV1()
+
+	// create rbac
+	kubectlWrapper("create", "deploy/rbac.yaml")
+	fmt.Println("Created rbac")
+
+	// create crd
+	kubectlWrapper("create", "deploy/crd.yaml")
+	fmt.Println("Created crd")
+
+	// create operator
+	kubectlWrapper("create", "deploy/operator.yaml")
+	fmt.Println("Created operator")
+
+	deploymentReplicaCheck(api, "memcached-operator", 1, 60)
+
+	// create example-memcached yaml file
+	file, err := os.OpenFile("deploy/cr.yaml", os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+
+	_, err = file.WriteString("apiVersion: \"cache.example.com/v1alpha1\"\nkind: \"Memcached\"\nmetadata:\n  name: \"example-memcached\"\nspec:\n  size: 3")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	file.Close()
+
+	// apply/create example-memcached deployment
```

Comment:
We're actually creating the Memcached Custom Resource(CR). Not the deployment itself.
So you can update or remove this comment.